npm/src/upload/upload-form.ts

```diff
@@ -99,13 +99,18 @@ export class UploadForm {
   }
```

Comment:
It might be better to add the consignmentId and parentFolder values to an object, and pass this through to the other functions, instead of passing them as separate params:
1. This will reduce the number of params being passed to functions;
2. If for example in the future we allow users to upload more than one folder at a time, just need to change the object to accommodate this
3. If we need to pass through other information taken from the upload event in the future, just need to change the object, by adding a new fields for the other information.

Would suggest a new interface in this file (not sure about the name):

```
export interface IUploadFileInfo {
  consignmentId: string;  
  parentFolder: string;  
}
```
This can then be passed around, and referenced by the other functions as needed, for example:

```
...
addSubmitListener(
    uploadFiles: (
      files: TdrFile[],
      uploadFileInfo:  IUploadFileInfo
    ) => void
  ) {
    this.formElement.addEventListener("submit", (ev) => {
      ev.preventDefault()
      const target: HTMLInputTarget | null = ev.currentTarget
      const files = this.retrieveFiles(target)
      const parentFolder = this.getParentFolderName(files)
      const uploadFileInfo: IUploadFileInfo = {
        consignmentId: this.consignmentId(),
        parentFolder: parentFolder
      }

      uploadFiles(files, uploadFileInfo)
    })
  }
...
```

Reply:
That makes much more sense yeah! I'll change it to do that :+1: